apps/avatax/src/modules/saleor/order.ts

```diff
@@ -47,3 +51,69 @@ export class SaleorOrder {
     return this.data.order.channel.taxConfiguration.pricesEnteredWithTax;
   }
 }
+
+interface ISaleorCancelledOrderPayload {
+  getChannelSlug(): string;
+  getAvataxId(): string;
+  getPrivateMetadata(): Array<{ key: string; value: string }>;
+}
+
+export class SaleorCancelledOrderEvent implements ISaleorCancelledOrderPayload {
+  private static schema = z.object({
+    order: z.object({
+      channel: z.object({
+        id: z.string(),
+        slug: z.string(),
+      }),
+      id: z.string(),
+      avataxId: z.string(),
+    }),
+    recipient: z.object({
+      privateMetadata: z.array(
+        z.object({
+          key: z.string(),
+          value: z.string(),
+        }),
+      ),
+    }),
+  });
+
+  private constructor(private data: z.infer<typeof SaleorCancelledOrderEvent.schema>) {}
+
+  getPrivateMetadata() {
+    return this.data.recipient.privateMetadata;
+  }
+
+  getChannelSlug() {
+    return this.data.order.channel.slug;
+  }
+
+  getAvataxId() {
+    return this.data.order.avataxId;
+  }
+
+  static create(payload: OrderCancelledPayloadFragment) {
+    if (!payload.order) {
+      return err(new OrderCancelPayloadOrderError("Insufficient order data"));
+    }
+
+    if (!payload.order.avataxId) {
+      return err(new OrderCancelNoAvataxIdError("No AvaTax id found in order"));
+    }
+
+    const ParsingError = BaseError.subclass("AvataxAppSaleorOrderCancelledParsingError");
```

Comment:
It should be exported or be static member of this class. We should be able to use this `create()` and then

```
if(error instanceOf SaleorCancelledOrderEvent.ParsingError) {
 // react on that
}
```